File: app/src/main/java/com/linderaredux/api/response/patient/Patient.java

package com.linderaredux.api.response.patient;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Patient implements Parcelable {

    @SerializedName("_id")
    @Expose
    public String id;
    @SerializedName("age")
    @Expose
    public Integer age;
    @SerializedName("height")
    @Expose
    public Integer height;
    @SerializedName("weight")
    @Expose
    public Integer weight;
    @SerializedName("hasDementia")
    @Expose
    public Boolean hasDementia;
    @SerializedName("residential_form")
    @Expose
    public String residentialForm;
    @SerializedName("firstname")
    @Expose
    public String firstname;
    @SerializedName("lastname")
    @Expose
    public String lastname;
    @SerializedName("sex")
    @Expose
    public String sex;
    @SerializedName("userID")
    @Expose
    public String userID;
    @SerializedName("analyse")
    @Expose
    public List<Analyse> analyse = null;
    @SerializedName("answerquestionaire")
    @Expose
    public List<Answerquestionaire> answerquestionaire = null;
    @SerializedName("answerquestionnaire")
    @Expose
    public List<Answerquestionnaire> answerquestionnaire = null;
    @SerializedName("video")
    @Expose
    public List<Video> video = null;
    public final static Creator<Patient> CREATOR = new Creator<Patient>() {


        @SuppressWarnings({
                "unchecked"
        })
        public Patient createFromParcel(Parcel in) {
            return new Patient(in);
        }

        public Patient[] newArray(int size) {
            return (new Patient[size]);
        }

    };

    protected Patient(Parcel in) {
        this.id = ((String) in.readValue((String.class.getClassLoader())));
        this.age = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.height = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.weight = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.hasDementia = ((Boolean) in.readValue((Boolean.class.getClassLoader())));
        this.residentialForm = ((String) in.readValue((String.class.getClassLoader())));
        this.firstname = ((String) in.readValue((String.class.getClassLoader())));
        this.lastname = ((String) in.readValue((String.class.getClassLoader())));
        this.sex = ((String) in.readValue((String.class.getClassLoader())));
        this.userID = ((String) in.readValue((String.class.getClassLoader())));
        in.readList(this.analyse, (Analyse.class.getClassLoader()));
        in.readList(this.answerquestionaire, (Answerquestionaire.class.getClassLoader()));
        in.readList(this.answerquestionnaire, (Answerquestionnaire.class.getClassLoader()));
        in.readList(this.video, (Video.class.getClassLoader()));
    }

    public Patient() {
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
        dest.writeValue(age);
        dest.writeValue(height);
        dest.writeValue(weight);
        dest.writeValue(hasDementia);
        dest.writeValue(residentialForm);
        dest.writeValue(firstname);
        dest.writeValue(lastname);
        dest.writeValue(sex);
        dest.writeValue(userID);
        dest.writeList(analyse);
        dest.writeList(answerquestionaire);
        dest.writeList(answerquestionnaire);
        dest.writeList(video);
    }

    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/com/linderaredux/api/response/patient/Patient.kt
package com.linderaredux.api.response.patient

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "patient")
data class Patient(@PrimaryKey(autoGenerate = true)
                   private var pid: Int = 0,
                   @SerializedName("firstname")
                   val firstname: String?,
                   @SerializedName("sex")
                   val sex: String?,
                   @SerializedName("analyse")
                   val analyse: List<Analyse>?,
                   @SerializedName("weight")
                   val weight: Int,
                   @SerializedName("video")
                   val video: List<Video>?,
                   @SerializedName("userID")
                   val userID: String?,
                   @SerializedName("lastname")
                   val lastname: String?,
                   @SerializedName("hasDementia")
                   val hasDementia: Boolean = false,
                   @SerializedName("answerquestionnaire")
                   val answerquestionnaire: List<Answerquestionnaire>?,
                   @SerializedName("answerquestionaire")
                   val answerquestionaire: List<Answerquestionaire>?,
                   @SerializedName("_id")
                   val Id: String?,
                   @SerializedName("residential_form")
                   val residentialForm: String?,
                   @SerializedName("age")
                   val age: Int = 0,
                   @SerializedName("height")
                   val height: Int = 0) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            parcel.createTypedArrayList(Analyse),
            parcel.readInt(),
            parcel.createTypedArrayList(Video),
            parcel.readString(),
            parcel.readString(),
            parcel.readByte() != 0.toByte(),
            parcel.createTypedArrayList(Answerquestionnaire),
            parcel.createTypedArrayList(Answerquestionaire),
            parcel.readString(),
            parcel.readString(),
            parcel.readInt(),
            parcel.readInt())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(pid)
        parcel.writeString(firstname)
        parcel.writeString(sex)
        parcel.writeTypedList(analyse)
        parcel.writeInt(weight)
        parcel.writeTypedList(video)
        parcel.writeString(userID)
        parcel.writeString(lastname)
        parcel.writeByte(if (hasDementia) 1 else 0)
        parcel.writeTypedList(answerquestionnaire)
        parcel.writeTypedList(answerquestionaire)
        parcel.writeString(Id)
        parcel.writeString(residentialForm)
        parcel.writeInt(age)
        parcel.writeInt(height)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Patient> {
        override fun createFromParcel(parcel: Parcel): Patient {
            return Patient(parcel)
        }

        override fun newArray(size: Int): Array<Patient?> {
            return arrayOfNulls(size)
        }
    }
}